src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -601,12 +601,15 @@ class RemoteMachineTrainingService implements TrainingService {
         let command: string;
         // Set CUDA_VISIBLE_DEVICES environment variable based on cuda_visible_device
         // If no valid cuda_visible_device is defined, set CUDA_VISIBLE_DEVICES to empty string to hide GPU device
-        if (typeof cuda_visible_device === 'string' && cuda_visible_device.length > 0) {
-            command = `CUDA_VISIBLE_DEVICES=${cuda_visible_device} ${this.trialConfig.command}`;
+        if (this.trialConfig.gpuNum === undefined) {
```

Comment:
Add comments to explain if gpuNum is undefined we will not set CUDA_VISIBLE_DEVICE?